File: backend/app/models/service.rb
class Service < ApplicationRecord
  has_many :service_requests
  has_one_attached :service_image

  # Validations
  validates :name, presence: true, uniqueness: true
  validates :base_price, numericality: { greater_than_or_equal_to: 0 }, allow_nil: true
  validates :estimated_duration, numericality: { greater_than: 0 }, allow_nil: true
  validates :category, inclusion: { in: %w[maintenance repair diagnostic inspection bodywork other general] }, allow_nil: true

  # Default scope - order by name alphabetically
  default_scope { order(name: :asc) }

  # Scopes for common queries
  scope :active, -> { where(active: true) }
  scope :inactive, -> { where(active: false) }
  scope :with_pricing, -> { where.not(base_price: nil) }
  scope :by_category, ->(category) { where(category: category) }
  scope :popular, -> { joins(:service_requests).group('services.id').order('COUNT(service_requests.id) DESC') }

  # Instance Methods

  # Returns formatted price string (e.g., "$120.00")
  def formatted_price
    base_price ? "$#{base_price.round(2)}" : "Quote Required"
  end

  # Returns true if this service has a set price
  def has_fixed_price?
    base_price.present?
  end

  # Returns count of service requests for this service
  def request_count
    service_requests.count
  end
end

File: frontend/src/types/service.ts
/* ==========================================
   SERVICE TYPES
   ========================================== */

export interface Service {
  id: number;
  name: string;
  description: string | null;
  detailed_description: string | null;
  base_price: string | null;
  estimated_duration: number | null;
  category: string;
  active: boolean;
  service_image_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface ServiceFormData {
  name: string;
  description: string;
  detailed_description: string;
  base_price: number | string;
  estimated_duration: number | string;
  category: string;
  active: boolean;
}

export const SERVICE_CATEGORIES = [
  'general',
  'maintenance',
  'repair',
  'diagnostic',
  'inspection',
  'bodywork',
  'other'
] as const;

export type ServiceCategory = typeof SERVICE_CATEGORIES[number];
